searchcore/src/tests/proton/documentdb/lid_space_compaction/lid_space_common.cpp

```diff
@@ -0,0 +1,223 @@
+        _validItr = false;
+    }
+    return search::DocumentMetaData();
+}
+search::DocumentMetaData MyScanIterator::getMetaData(uint32_t lid) const {
+    return search::DocumentMetaData(lid, TIMESTAMP_1, createBucketId(lid), GID_1);
+}
+
+document::BucketId
+MyScanIterator::createBucketId(uint32_t lid) const {
+    return _bucketIdEqualLid ? document::BucketId(lid) : BUCKET_ID_1;
+}
+
+void
+MyHandler::clearMoveDoneContexts() {
+    _moveDoneContexts.clear();
+}
+void
+MyHandler::run_remove_ops(bool remove_batch) {
+    // This ensures to max out the threshold time in the operation rate tracker.
+    if (remove_batch) {
+        _op_listener->notify_remove_batch();
+        _op_listener->notify_remove_batch();
+        _op_listener->notify_remove_batch();
+    } else {
+        _op_listener->notify_remove();
+        _op_listener->notify_remove();
+        _op_listener->notify_remove();
+    }
+}
+void
+MyHandler::stop_remove_ops(bool remove_batch) const {
+    if (remove_batch) {
+        _rm_listener->get_remove_batch_tracker().reset(vespalib::steady_clock::now());
+    } else {
+        _rm_listener->get_remove_tracker().reset(vespalib::steady_clock::now());
+    }
+}
+vespalib::string
+MyHandler::getName() const {
+    return "myhandler";
+}
+void
+MyHandler::set_operation_listener(documentmetastore::OperationListener::SP op_listener) {
+    auto* rm_listener = dynamic_cast<RemoveOperationsRateTracker*>(op_listener.get());
+    assert(rm_listener != nullptr);
+    _op_listener = std::move(op_listener);
+    _rm_listener = rm_listener;
+}
+LidUsageStats
+MyHandler::getLidStatus() const {
+    assert(_handleMoveCnt < _stats.size());
+    return _stats[_handleMoveCnt];
+}
+IDocumentScanIterator::UP
+MyHandler::getIterator() const {
+    assert(_iteratorCnt < _lids.size());
+    return std::make_unique<MyScanIterator>(_lids[_iteratorCnt++], _bucketIdEqualLid);
+}
+MoveOperation::UP
+MyHandler::createMoveOperation(const search::DocumentMetaData &document, uint32_t moveToLid) const {
+    assert(document.lid > moveToLid);
+    _moveFromLid = document.lid;
+    _moveToLid = moveToLid;
+    return std::make_unique<MoveOperation>();
+}
+void
+MyHandler::handleMove(const MoveOperation &, IDestructorCallback::SP moveDoneCtx) {
+    ++_handleMoveCnt;
+    if (_storeMoveDoneContexts) {
+        _moveDoneContexts.push_back(std::move(moveDoneCtx));
+    }
+}
+void
+MyHandler::handleCompactLidSpace(const CompactLidSpaceOperation &op, std::shared_ptr<IDestructorCallback>) {
+    _wantedLidLimit = op.getLidLimit();
+}
+
+MyHandler::MyHandler(bool storeMoveDoneContexts, bool bucketIdEqualLid)
+    : _stats(),
+      _moveFromLid(0),
+      _moveToLid(0),
+      _handleMoveCnt(0),
+      _wantedLidLimit(0),
+      _iteratorCnt(0),
+      _storeMoveDoneContexts(storeMoveDoneContexts),
+      _bucketIdEqualLid(bucketIdEqualLid),
+      _moveDoneContexts(),
+      _op_listener(),
+      _rm_listener()
+{}
+
+MyHandler::~MyHandler() = default;
+
+#if 0
```

Comment:
Is this intentional? Consider removing the code block if not used.

Reply:
No, cleaned up now.